logic/src/commonMain/kotlin/com/wire/kalium/logic/GenericFailure.kt

```diff
@@ -0,0 +1,19 @@
+package com.wire.kalium.logic
+
+sealed class GenericFailure {
+    /**
+     * Failed to establish a connection with the necessary servers in order to pull/push data.
+     * Caused by weak - complete lack of - internet connection.
+     */
+    object NoNetworkConnection : GenericFailure()
+
+    /**
+     * Server internal error, or we can't parse the response,
+     * or anything API-related that is out of control from the user.
+     * Either fix our app or our backend.
+     */
+    object ServerMiscommunication : GenericFailure()
+
+    class UnknownFailure(val rootCause: Throwable?): GenericFailure()
+
+}
```

Comment:
Could we somehow reuse the `GenericError` class from the network module?

Reply:
We could. Personaly, I don't think we should.

`GenericError` from network module are exactly that, generic errors from **network** module.

If, for example, we have generic DB errors (like a missing migration, or corrupted db file), the **network** errors have nothing to do with that.

And even if we forget about DB and other kinds of data sources, some of the `BadRequest` and `Serialization` errors in `Network` can be treated as `generic` from of the business logic. So some kind of mapping is needed anyway. For example:

Serialization error? Bad Request? Server returning 500? Our users don't know, they don't care. They all are `ServerMiscommunication` and they all mean something along the lines of "Our services are not working properly now, please try again later".

From what I see, this is business logic, this is how we should treat users. And the logic for errors should be completely detached from the network or any other data source.